### ISSUE TITLE
`divided_by` filter incorrectly rounds down when divisor is float

### ISSUE THREAD

**Issue by kennytm** [cobalt-org/liquid-rust#88]
[According to documentation](https://shopify.github.io/liquid/filters/divided_by/), `divided_by` floors only when the divisor is an integer. Otherwise it should not round.

```
{{ 20 | divided_by: 7 }}
⇒ 2

{{ 20 | divided_by: 7.0 }}
⇒ 2.857142857142857

{% assign my_integer = 7 %}
{{ 20 | divided_by: my_integer }}
⇒ 2

{% assign my_integer = 7 %}
{% assign my_float = my_integer | times: 1.0 %}
{{ 20 | divided_by: my_float }}
⇒ 2.857142857142857
```

However, as of 0.9.1, all of these produce `2` as result since [it floors unconditionally](https://github.com/cobalt-org/liquid-rust/blob/73d480075a70f40028cdb41034ae601cf8366e70/src/filters.rs#L221).

I think you may need to split `Value::Num` into two distinct variants to properly fix it.

**Comment by epage** [comment]
Thanks for catching this.

Yeah, this will probably require integer support and we should probably audit the rest of the filters for when we should be returning integer or decimal values.

**Comment by epage** [comment]
Fixed by #158
